### PR TITLE
Update README for CORS and toggling session_service

### PR DESCRIPTION
**Pull request by jopecko**
* add description of CORS configuration
* add example of toggling session_service on/off for route
* rename config var access_control_request_headers to allowed_http_headers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,6 +82,58 @@ Once you have written your routes, a good step is to run the `mix validate_route
...
+
+CORS is a protocol that allows web applications to make requests from a browser
+across different domains. A full specification can be found [here](https://www.w3.org/TR/cors/).
+
+#### Configuring the CORS filter
+
+The middleware can be configured within the routes configuration file. For a full
+listing of configuration options, see the [config.exs](blob/master/config/config.exs#L70-L84).
+
+The available options include:
```
Comment (MichaelSitter) [inline README.md:97]: For the options listed below should we be providing examples? Or is the intended usage clear enough in the context of cors support?

File: README.md [hunk 2]
```diff
@@ -82,6 +82,58 @@ Once you have written your routes, a good step is to run the `mix validate_route
...
+* allowed_origins - allow only requests with origins from a whitelist (by default all origins are allowed)
+* allowed_http_methods - allow only HTTP methods from a whitelist for preflight requests (by default all methods are allowed)
+* allowed_http_headers - allow only HTTP headers from a whitelist for preflight requests (by default all headers are allowed)
+* preflight_max_age - set how long the results of a preflight request can be cached in a preflight result cache (by default 1 hour)
+* allowed_origins - enable/disable serving requests with origins not in whitelist as non-CORS requests (by default they are forbidden)
+* any_origin_allowed - enable/disable serving requests with origins not in whitelist as non-CORS requests (by default they are forbidden)
+
+### AuthN Middleware with Session Service Integration
+
+Spacesuit provide an AuthN middleware, coupled with a session-providing service, that **when
```
Comment (MichaelSitter) [inline README.md:109]: provides

File: config/config.exs [hunk 3]
```diff
@@ -70,17 +70,19 @@ config :spacesuit, session_service: %{ enabled: false }
 # Configuration for the CORS middleware
 config :spacesuit, cors: %{
   # Main kill switch to disable the middleware
-  enabled: false, 
-  # Required prefix match in order to process CORS
-  #path_prefixes: ["/matched"],
-  # Maximum age for preflight requests
+  enabled: false,
+  # The path prefixes to filter.
```
Comment (MichaelSitter) [inline config/config.exs:74]: This description actually confused me. The intent is, 'any requests which match this prefix should be processed as CORS requests', correct?

File: README.md (hunk 1 above)
Comment (jopecko) [reply README.md:97]: think @relistan prefers using the commented out code in the `config.exs` to serve as example configuration; i've called that out just above this.

File: config/config.exs (hunk 3 above)
Comment (jopecko) [reply config/config.exs:74]: how about `The matching path prefixes to process as CORS requests`?

**Pull request closed without merging**